mitodl/open-discussions
# Use React 18 in New Pages

#### Pre-Flight checklist


#### What are the relevant tickets?
#3990 

#### What's this PR do?
This PR updates React to version 18 **in infinite-corridor only**. I.e., "old open" is not affected. 

Now:
```
frontends:
|-- infinite-corridor      <-- React 18 as dependency
|-- ol-forms                      ... (React is peer dependency)
|-- ol-search-ui                  ... (React is peer dependency)
|-- ol-util                       ... (React is peer dependency)
|-- ol-widgets                    ... (React is peer dependency)
|-- open-discussions       <-- React 16 as dependency
```

#### How should this be manually tested?
1. Stop docker
2. `docker compose up`; wait for installs to finish.
3. view "old open" pages and "new open" pages (/infinite, /infinite/search, /infinite/lists). They should look and function the same.

#### Where should the reviewer start?
I would recommend starting with the [comment on `hoistingLimits`](https://github.com/mitodl/open-discussions/pull/4007/files#r1204357894)
